replit/toml-editor
# Table Header + Dotted Path Add

# Why?

https://linear.app/replit/issue/DX-523/toml-editor-add-ability-to-specify-a-tableheaderpath-plus-a-dottedpath

## Changes

1. added `table_header_path` and `dotted_path` fields to `Op`
2. implemented `add_value_with_table_header_and_dotted_path` function in table_header_adder.rs
3. use `add_value_with_table_header_and_dotted_path` if `table_header_path` is supplied

## Test

1. Make input.json file:
```
[{"op":"add","path":"","table_header_path":"moduleConfig","dotted_path":"bundles/typecript/enable","value":"true"}]
```

2. create empty file `test.toml`
3. `cat input.json | cargo run -- -p test.toml`
4. `test.toml` should contain:
```
[moduleConfig]
bundles.typecript.enable = true
```
